### PR TITLE
Multiplayer.network.items

### PR DESCRIPTION
**Pull request by herman-rogers**
Add multiplayer smooth syncing to paperball items. Change the way the spawn system and item system fundamentally work. Update the mandrake item and add in the older Sprite based UI system.

### CODE REVIEW

File: Assets/Scripts/Inventory/ItemType/WeaponItem.cs [hunk 1]
```diff
@@ -5,34 +5,30 @@ public class WeaponItem : Item {
...
-                spawnVisual.SetActive( false );
-                activeVisual.SetActive( false );
-                break;
-            case PlayerInventory.REMOVED_ITEM_FROM_INVENTORY:
-                currentItemState = ItemState.ITEM_AT_SPAWN_POINT;
-                weaponItemData.CostOfUse = cachedCostOfUse;
-                break;
-        }
-    }
+    //public override void OnNotify( Object sender, EventArguments e ) {
```
Comment (chvck) [inline Assets/Scripts/Inventory/ItemType/WeaponItem.cs]: quite a few comments, WIP?

File: Assets/Scripts/Inventory/Items/MandrakeItem.cs [hunk 2]
```diff
@@ -5,49 +5,72 @@
...
+        NetworkServer.Destroy( gameObject );
     }
 
-    [Command]
-    private void CmdPlaceMandrakeInPlayerHands( GameObject player ) {
-        GameObject playerHands = player.GetComponentInChildren<PlayerHands>( ).gameObject;
-        gameObject.transform.position = playerHands.transform.position;
-        gameObject.transform.parent = playerHands.transform;
+    [ClientRpc]
+    private void RpcMakePlayerInvisible( GameObject player ) {
```
Comment (chvck) [inline Assets/Scripts/Inventory/Items/MandrakeItem.cs]: These 2 could just be refactored into 1 set player visible or something

File: Assets/Scripts/Observer/UnityObserver.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 public class UnityObserver : NetworkBehaviour {
     //Don't Override Awake
-    protected void Awake( ) {
+    protected virtual void Awake( ) {
```
Comment (chvck) [inline Assets/Scripts/Observer/UnityObserver.cs]: Does this need to happy as we have that overridable InitializeUnityObserver?

File: Assets/Scripts/Inventory/PlayerInventory.cs [hunk 4]
```diff
@@ -1,77 +1,116 @@
...
+                CmdInitializeNewWeaponItem( knifeItem );
                 break;
         }
     }
 
-    [Command]
-    private void CmdRemoveThrownItem( ) {
-        Notify( ITEM_THROWN_BY_PLAYER );
-        CmdRemoveItemFromInventory( );
+    private void InitializeNewDefenseItem( GDEDefenseItemData item ) {
```
Comment (chvck) [inline Assets/Scripts/Inventory/PlayerInventory.cs]: Think these 2 can be rolled into one using <Type> or something

File: Assets/Scripts/Observer/UnityObserver.cs (hunk 3 above)
Comment (herman-rogers) [inline Assets/Scripts/Observer/UnityObserver.cs]: It was necessary before, but then I removed it and did other things so not any more

File: Assets/Scripts/Inventory/Items/MandrakeItem.cs (hunk 2 above)
Comment (chvck) [inline Assets/Scripts/Inventory/Items/MandrakeItem.cs]: Yup

File: Assets/Scripts/Inventory/PlayerInventory.cs (hunk 4 above)
Comment (chvck) [inline Assets/Scripts/Inventory/PlayerInventory.cs]: Actually, not sure it can be

Comment (herman-rogers) [inline Assets/Scripts/Inventory/PlayerInventory.cs]: I'll take it as a challenge.

**Pull request merged**